cmd/kubeadm/app/util/runtime/runtime.go

```diff
@@ -173,3 +174,19 @@ func detectCRISocketImpl(isSocket func(string) bool, knownCRISockets []string) (
 func DetectCRISocket() (string, error) {
 	return detectCRISocketImpl(isExistingSocket, defaultKnownCRISockets)
 }
+
+// SandboxImage returns the sandbox image used by the container runtime
+func (runtime *CRIRuntime) SandboxImage() (string, error) {
+	args := []string{"info", "-o", "go-template", "--template", "{{.config.sandboxImage}}"}
```

Comment:
```
[root@paco-centos-9 ~]# crictl --debug=false info -o go-template --template '{{.config.sandboxImage}}'
m.daocloud.io/registry.k8s.io/pause:3.9
[root@paco-centos-9 ~]# cat /etc/crictl.yaml
runtime-endpoint: unix:///var/run/containerd/containerd.sock
image-endpoint: unix:///var/run/containerd/containerd.sock
timeout: 15
debug: true
pull-image-on-create: false
[root@paco-centos-9 ~]# crictl info -o go-template --template '{{.config.sandboxImage}}'
DEBU[0000] get runtime connection
DEBU[0000] StatusRequest: &StatusRequest{Verbose:true,}
DEBU[0000] StatusResponse: &StatusResponse{Status:&RuntimeStatus{Conditions:[]*RuntimeCondition{&RuntimeCondition{Type:RuntimeReady,Status:true,Reason:,Message:,},&RuntimeCondition{Type:NetworkReady,Status:true,Reason:,Message:,},},},Info:map[string]string{cniconfig: {"PluginDirs":["/opt/cni/bin"],"PluginConfDir":"/etc/cni/net.d","PluginMaxConfNum":1,"Prefix":"eth","Networks":[{"Config":{"Name":"cni-loopback","CNIVersion":"0.3.1","Plugins":[{"Network":{"type":"loopback","ipam":{},"dns":{}},"Source":"{\"type\":\"loopback\"}"}],"Source":"{\n\"cniVersion\": \"0.3.1\",\n\"name\": \"cni-loopback\",\n\"plugins\": [{\n  \"type\": \"loopback\"\n}]\n}"},"IFName":"lo"},{"Config":{"Name":"k8s-pod-network","CNIVersion":"0.3.1","Plugins":[{"Network":{"type":"calico","ipam":{"type":"calico-ipam"},"dns":{}},"Source":"{\"container_settings\":{\"allow_ip_forwarding\":false},\"datastore_type\":\"kubernetes\",\"ipam\":{\"assign_ipv4\":\"true\",\"assign_ipv6\":\"false\",\"type\":\"calico-ipam\"},\"kubernetes\":{\"k8s_api_root\":\"https://10.96.0.1:443\",\"kubeconfig\":\"/etc/cni/net.d/calico-kubeconfig\"},\"log_file_path\":\"/var/log/calico/cni/cni.log\",\"log_level\":\"Info\",\"mtu\":0,\"nodename_file_optional\":false,\"policy\":{\"type\":\"k8s\"},\"type\":\"calico\"}"},{"Network":{"type":"bandwidth","capabilities":{"bandwidth":true},"ipam":{},"dns":{}},"Source":"{\"capabilities\":{\"bandwidth\":true},\"type\":\"bandwidth\"}"},{"Network":{"type":"portmap","capabilities":{"portMappings":true},"ipam":{},"dns":{}},"Source":"{\"capabilities\":{\"portMappings\":true},\"snat\":true,\"type\":\"portmap\"}"}],"Source":"{\n  \"name\": \"k8s-pod-network\",\n  \"cniVersion\": \"0.3.1\",\n  \"plugins\": [\n    {\n      \"type\": \"calico\",\n      \"datastore_type\": \"kubernetes\",\n      \"mtu\": 0,\n      \"nodename_file_optional\": false,\n      \"log_level\": \"Info\",\n      \"log_file_path\": \"/var/log/calico/cni/cni.log\",\n      \"ipam\": { \"type\": \"calico-ipam\", \"assign_ipv4\" : \"true\", \"assign_ipv6\" : \"false\"},\n      \"container_settings\": {\n          \"allow_ip_forwarding\": false\n      },\n      \"policy\": {\n          \"type\": \"k8s\"\n      },\n      \"kubernetes\": {\n          \"k8s_api_root\":\"https://10.96.0.1:443\",\n          \"kubeconfig\": \"/etc/cni/net.d/calico-kubeconfig\"\n      }\n    },\n    {\n      \"type\": \"bandwidth\",\n      \"capabilities\": {\"bandwidth\": true}\n    },\n    {\"type\": \"portmap\", \"snat\": true, \"capabilities\": {\"portMappings\": true}}\n  ]\n}"},"IFName":"eth0"}]},config: {"containerd":{"snapshotter":"overlayfs","defaultRuntimeName":"runc","defaultRuntime":{"runtimeType":"","runtimePath":"","runtimeEngine":"","PodAnnotations":[],"ContainerAnnotations":[],"runtimeRoot":"","options":{},"privileged_without_host_devices":false,"baseRuntimeSpec":"","cniConfDir":"","cniMaxConfNum":0},"untrustedWorkloadRuntime":{"runtimeType":"","runtimePath":"","runtimeEngine":"","PodAnnotations":[],"ContainerAnnotations":[],"runtimeRoot":"","options":{},"privileged_without_host_devices":false,"baseRuntimeSpec":"","cniConfDir":"","cniMaxConfNum":0},"runtimes":{"runc":{"runtimeType":"io.containerd.runc.v2","runtimePath":"","runtimeEngine":"","PodAnnotations":[],"ContainerAnnotations":[],"runtimeRoot":"","options":{"BinaryName":"","CriuImagePath":"","CriuPath":"","CriuWorkPath":"","IoGid":0,"IoUid":0,"NoNewKeyring":false,"NoPivotRoot":false,"Root":"","ShimCgroup":"","SystemdCgroup":true},"privileged_without_host_devices":false,"baseRuntimeSpec":"","cniConfDir":"","cniMaxConfNum":0}},"noPivot":false,"disableSnapshotAnnotations":true,"discardUnpackedLayers":false,"ignoreRdtNotEnabledErrors":false},"cni":{"binDir":"/opt/cni/bin","confDir":"/etc/cni/net.d","maxConfNum":1,"confTemplate":"","ipPref":""},"registry":{"configPath":"","mirrors":{},"configs":{},"auths":{},"headers":{"User-Agent":["containerd/v1.6.15"]}},"imageDecryption":{"keyModel":"node"},"disableTCPService":true,"streamServerAddress":"127.0.0.1","streamServerPort":"0","streamIdleTimeout":"4h0m0s","enableSelinux":false,"selinuxCategoryRange":1024,"sandboxImage":"m.daocloud.io/registry.k8s.io/pause:3.9","statsCollectPeriod":10,"systemdCgroup":false,"enableTLSStreaming":false,"x509KeyPairStreaming":{"tlsCertFile":"","tlsKeyFile":""},"maxContainerLogSize":16384,"disableCgroup":false,"disableApparmor":false,"restrictOOMScoreAdj":false,"maxConcurrentDownloads":3,"disableProcMount":false,"unsetSeccompProfile":"","tolerateMissingHugetlbController":true,"disableHugetlbController":true,"device_ownership_from_security_context":false,"ignoreImageDefinedVolumes":false,"netnsMountsUnderStateDir":false,"enableUnprivilegedPorts":false,"enableUnprivilegedICMP":false,"containerdRootDir":"/var/lib/containerd","containerdEndpoint":"/run/containerd/containerd.sock","rootDir":"/var/lib/containerd/io.containerd.grpc.v1.cri","stateDir":"/run/containerd/io.containerd.grpc.v1.cri"},golang: "go1.18.9",lastCNILoadStatus: OK,lastCNILoadStatus.default: OK,},}
m.daocloud.io/registry.k8s.io/pause:3.9
```
```suggestion
	args := []string{"--debug=false", "info", "-o", "go-template", "--template", "{{.config.sandboxImage}}"}
```

Reply:
As I always debug things, I enabled the debug by configuration and I am not sure if this will encounter such problem when trim the result here.
- I have trouble testing it. Can you give a try? And if no problem with the debug messages, this do not need a change.

Reply:
Thanks @pacoxu 
Make sense. I have updated the command args.